pkg/ipmi/rakp_message_1.go

```diff
@@ -75,6 +75,36 @@ func (r *RAKPMessage1) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.Ser
 	return nil
 }
 
+func (*RAKPMessage1) NextLayerType() gopacket.LayerType {
+	return gopacket.LayerTypePayload
+}
+
+func (r *RAKPMessage1) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+	if len(data) < 28 { // minimum in case of non-zero status code
+		df.SetTruncated()
+		return fmt.Errorf("RAKP Message 2 must be at least 28 bytes, got %v", len(data))
+	}
+	r.BaseLayer.Contents = data
+	r.Tag = uint8(data[0])
+	r.ManagedSystemSessionID = binary.LittleEndian.Uint32(data[4:8])
+	copy(r.RemoteConsoleRandom[:], data[8:24])
+	r.MaxPrivilegeLevel = PrivilegeLevel(data[24] & 0xF)
+	r.PrivilegeLevelLookup = (data[24] & (1 << 4)) == 0
+	lenUsername := data[27]
```

Comment:
Given this is a numeric length, should we convert to `uint8`?

Reply:
You are right.